supportedBackends/tes/src/main/scala/cromwell/backend/impl/tes/TesAsyncBackendJobExecutionActor.scala

```diff
@@ -0,0 +1,206 @@
+import cromwell.core.path.Path
+import cromwell.core.retry.SimpleExponentialBackoff
+import spray.client.pipelining._
+import spray.http.HttpRequest
+import spray.httpx.SprayJsonSupport._
+import spray.httpx.unmarshalling._
+import wdl4s.expression.NoFunctions
+import wdl4s.values.WdlFile
+
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+import scala.language.postfixOps
+import scala.util.{Failure, Success}
+
+sealed trait TesRunStatus {
+  def isTerminal: Boolean
+}
+
+case object Running extends TesRunStatus {
+  def isTerminal = false
+}
+
+case object Complete extends TesRunStatus {
+  def isTerminal = true
+}
+
+case object FailedOrError extends TesRunStatus {
+  def isTerminal = true
+}
+
+object TesAsyncBackendJobExecutionActor {
+  val JobIdKey = "tes_job_id"
+}
+
+class TesAsyncBackendJobExecutionActor(override val standardParams: StandardAsyncExecutionActorParams)
+  extends BackendJobLifecycleActor with StandardAsyncExecutionActor with TesJobCachingActorHelper {
+
+  override type StandardAsyncRunInfo = Any
+
+  override type StandardAsyncRunStatus = TesRunStatus
+
+  override lazy val pollBackOff = SimpleExponentialBackoff(
+    initialInterval = 1 seconds,
+    maxInterval = 5 minutes,
+    multiplier = 1.1
+  )
+
+  override lazy val executeOrRecoverBackOff = SimpleExponentialBackoff(
+    initialInterval = 3 seconds,
+    maxInterval = 30 seconds,
+    multiplier = 1.1
+  )
+
+  override lazy val retryable: Boolean = false
+
+  private val tesEndpoint = workflowDescriptor.workflowOptions.getOrElse("endpoint", tesConfiguration.endpointURL)
+
+  override lazy val jobTag = jobDescriptor.key.tag
+
+  private def pipeline[T: FromResponseUnmarshaller]: HttpRequest => Future[T] = sendReceive ~> unmarshal[T]
+
+  // Utility for converting a WdlValue so that the path is localized to the
+  // container's filesystem.
+  override def mapCommandLineWdlFile(wdlFile: WdlFile): WdlFile = {
+    val localPath = Paths.get(wdlFile.valueString).toAbsolutePath
+    localPath match {
+      case p if p.startsWith(tesJobPaths.DockerRoot) =>
+        val containerPath = p.pathAsString
+        WdlFile(containerPath)
+      case p if p.startsWith(tesJobPaths.callExecutionRoot) =>
+        val containerPath = tesJobPaths.containerExec(commandDirectory, localPath.getFileName.pathAsString)
+        WdlFile(containerPath)
+      case p =>
+        val containerPath = tesJobPaths.containerInput(p.pathAsString)
+        WdlFile(containerPath)
+    }
+  }
+
+  override lazy val commandDirectory: Path = {
+    runtimeAttributes.dockerWorkingDir match {
+      case Some(path) => Paths.get(path)
+      case None => tesJobPaths.callExecutionDockerRoot
+    }
+  }
+
+  def createTaskMessage(): TesTaskMessage = {
+    val task = TesTask(jobDescriptor, configurationDescriptor, jobLogger, tesJobPaths, runtimeAttributes, commandDirectory, backendEngineFunctions)
+
+    tesJobPaths.script.write(commandScriptContents)
```

Comment:
Perhaps, as Miguel pointed out, changing the permissions might help, specifically here?

```scala
tesJobPaths.script.write(commandScriptContents).chmod("rwxrwxrwx")
```

I haven't tried running this myself yet though. I'm assuming that this is the right spot to fix permissions, based on the error message in the logs producing the [exit code 126](https://docs.docker.com/engine/reference/run/#/exit-status):

> /bin/bash: /cromwell-executions/woot/a2c488a2-e2a0-46f4-8714-47ac3a89bc2b/call-notroot/execution/script: Permission denied

I'm not 100% sure at the moment where/why the permissions on the script are not-as-expected though. The local backend gets around permissions errors by telling docker to run `/bin/bash /path/to/script`, that seems to work even if the file is `rw-rw-r--`.

If you really are running into problems and don't want to support this test and the "running as non-root" functionality for whatever reason, let us know, and we can figure out a way to exclude it.